src/main/java/com/google/api/codegen/transformer/go/GoGapicSurfaceTransformer.java

```diff
@@ -0,0 +1,258 @@
+    Map<String, ImmutableSet<Code>> retryCodesDef =
+        context.getInterfaceConfig().getRetryCodesDefinition();
+    Map<String, RetrySettings> retryParamsDef =
+        context.getInterfaceConfig().getRetrySettingsDefinition();
+    for (RetryPairDefinitionView.Name name : retryNames) {
+      ImmutableSet<Code> codes = retryCodesDef.get(name.codes());
+      if (codes.isEmpty()) {
+        continue;
+      }
+      ImmutableSet.Builder<String> retryCodeNames = ImmutableSet.<String>builder();
+      for (Code code : codes) {
+        retryCodeNames.add(namer.getStatusCodeName(code));
+      }
+      retryDef.put(
+          name,
+          RetryPairDefinitionView.newBuilder()
+              .name(name)
+              .retryCodes(retryCodeNames.build())
+              .params(retryParamsDef.get(name.params()))
+              .build());
+    }
+    return retryDef.values();
+  }
+
+  private Set<RetryPairDefinitionView.Name> getRetryNames(SurfaceTransformerContext context) {
+    Set<RetryPairDefinitionView.Name> retryNames = new HashSet<>();
+    for (Method method : context.getNonStreamingMethods()) {
+      MethodConfig conf = context.getMethodConfig(method);
+      retryNames.add(
+          RetryPairDefinitionView.Name.create(
+              conf.getRetrySettingsConfigName(), conf.getRetryCodesConfigName()));
+    }
+    return retryNames;
+  }
+
+  private static final String EMPTY_PROTO_PKG = "github.com/golang/protobuf/ptypes/empty";
+
+  private List<String> getImports(SurfaceTransformerContext context, boolean hasRetry) {
+    TreeSet<GoImport> standard = new TreeSet<>();
+    TreeSet<GoImport> thirdParty = new TreeSet<>();
+
+    standard.add(GoImport.create("fmt"));
+    standard.add(GoImport.create("runtime"));
+    if (hasRetry) {
+      standard.add(GoImport.create("time"));
+      thirdParty.add(GoImport.create("google.golang.org/grpc/codes"));
+    }
+    for (Method method : context.getNonStreamingMethods()) {
+      if (context.getMethodConfig(method).isPageStreaming()) {
+        standard.add(GoImport.create("math"));
+        break;
+      }
+    }
+    thirdParty.add(GoImport.create("golang.org/x/net/context"));
+    thirdParty.add(GoImport.create("google.golang.org/grpc"));
+    thirdParty.add(GoImport.create("google.golang.org/grpc/metadata"));
+    thirdParty.add(GoImport.create("github.com/googleapis/gax-go", "gax"));
+    thirdParty.add(GoImport.create("google.golang.org/api/option"));
+    thirdParty.add(GoImport.create("google.golang.org/api/transport"));
+
+    for (Map.Entry<String, String> entry : context.getTypeTable().getImports().entrySet()) {
+      if (!entry.getKey().equals(EMPTY_PROTO_PKG)) {
+        thirdParty.add(GoImport.create(entry.getKey(), entry.getValue()));
+      }
+    }
+
+    // We don't have to save import of empty proto.
+    // Instead of return the empty, we return nothing.
+
+    return formatImports(standard, thirdParty);
+  }
```

Comment:
I moved most of the logic into GoTypeTable. However, the `formatImports` function is still necessary since the idiom for imports in Go requires blank line between standard library and third party. Java needs to do this too right?